### DISCUSSION TITLE
¿Cómo puedo actualizar el icono de un canal que se encuentra en otra máquina de estado?

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#31] [2. Questions and Answers] [answered]
Entregaré más información

Tengo una SM 1 y creo 3 tasks: 1 a la SM 1, 1 a la SM 2 y 1 a la SM 3 y en las tasks de las SM 2 y SM 3 coloco el id de la task SM 1 en el campo Info.

En el estado n°3 de mi SM 1 quiero actualizar el ícono de las tasks SM 2 y SM 3, mediante una question de tipo imagen.

Mi pregunta es: ¿Cómo lo puedo hacer?

Gracias

**Comment by acedward** [comment]
@NicolasPLG Me imagino que te refieres a actualizar la imágen de un canal:

![image](https://user-images.githubusercontent.com/3951895/113592456-2bde7680-9603-11eb-9505-0c65b7c6e2fb.png)

* Primero tienes que encontrar el ID de tu canal:
  *   Si conoces solo las tareas puedes utlizar el campo task.channel
  *   Si conoces el canal, puedes utilizar su _id

* En segundo lugar debes subir la imagen y obtener su URL
* Tercero debes hacer un PATCH sobre el canal, puedes usar NWRequest a PATCH `/api/v2/channels/[id]` actualizando el campo que hace referencia a su ícono

e.g.
```
{
  image: {
    original: "https://cotalker-us.s3.amazonaws.com/image/1556119457262_original.jpeg"
    small: "https://cotalker-us.s3.amazonaws.com/image/1556119457260_small.jpeg"
    square: "https://cotalker-us.s3.amazonaws.com/image/1556119457260_square.jpeg"
  }
}
```

**Reply by NicolasPLG** [reply]
Hola @acedward 

Si es el icono de la task de un canal, tengo un problema no tengo la id de la task que creé. Entonces no sé como se puede obtener la task.

Otra opción que tengo es colocar la imagen cuando creo las tareas de las 2 sm (SM 2 y SM3). ¿Existe otra forma de hacerlo?

Gracias

**Comment by guillevalin** [comment, marked as answer]
@NicolasPLG el problema es que no tienes un endpoint para consultar fácilmente por las tareas 2 y 3 con el contexto de la tarea 1 que está en la SM1.

Te recomendaría asignar los íconos en el momento de creación, o de cambio de estado de aquellas tareas. En cada tarea tienes task.channel si haces un GET de la task por ID con el endpoint:
/api/tasks/[taskgroup]/task/[taskid]

A partir de ese channelId podrías usar PATCH de canales, con el endpoint, **/api/channels/[id]** usando como BODY lo que escribió @acedward en la respuesta anterior.

**Reply by acedward** [reply]
@NicolasPLG 
Otra opción es que al crear la tarea, **en la respuesta vendrá** en `task.channel` el _id del canal - donde puedes remplazar las imágenes.

E.g., si la etapa de creación se llama **crear_tarea_a** puedes en otro paso usar `$OUTPUT#crear_tarea_a#task|channel` para obtener el _id del canal
(No estoy 100% seguro si el argumento final es `task|channel`  o solo `channel` )

**Comment by NicolasPLG** [comment]
gracias a ambos @acedward y @guillevalin , voy a probar ambas opciones, cierro aqui.
